docs/modules/ROOT/pages/erc1155.adoc

```diff
@@ -8,104 +8,99 @@ The ERC1155 library implements an approximation of {eip-1155} in Cairo for Stark
+        owner: ContractAddress, operator: ContractAddress
     ) -> bool;
-    fn set_approval_for_all(ref self: TState, operator: ContractAddress, approved: bool);
-
-    // ISRC5
-    fn supports_interface(self: @TState, interface_id: felt252) -> bool;
+    fn set_approval_for_all( operator: ContractAddress, approved: bool);
 
     // IERC1155MetadataURI
-    fn uri(self: @TState, token_id: u256) -> ByteArray;
+    fn uri(token_id: u256) -> ByteArray;
+
+    // ISRC5
+    fn supports_interface(interface_id: felt252) -> bool;
 
     // IERC1155Camel
-    fn balanceOf(self: @TState, account: ContractAddress, tokenId: u256) -> u256;
+    fn balanceOf(account: ContractAddress, tokenId: u256) -> u256;
     fn balanceOfBatch(
-        self: @TState, accounts: Span<ContractAddress>, tokenIds: Span<u256>
+        accounts: Span<ContractAddress>, tokenIds: Span<u256>
     ) -> Span<u256>;
     fn safeTransferFrom(
-        ref self: TState,
         from: ContractAddress,
         to: ContractAddress,
         tokenId: u256,
         value: u256,
         data: Span<felt252>
     );
     fn safeBatchTransferFrom(
-        ref self: TState,
         from: ContractAddress,
         to: ContractAddress,
         tokenIds: Span<u256>,
         values: Span<u256>,
         data: Span<felt252>
     );
-    fn isApprovedForAll(self: @TState, owner: ContractAddress, operator: ContractAddress) -> bool;
-    fn setApprovalForAll(ref self: TState, operator: ContractAddress, approved: bool);
+    fn isApprovedForAll(owner: ContractAddress, operator: ContractAddress) -> bool;
+    fn setApprovalForAll(operator: ContractAddress, approved: bool);
 }
 ----
 
 === ERC1155 Compatibility
 
-Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC1155 standard
-using `u256` instead of `felt252`. But some differences can still be found, such as:
+Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC1155 standard but some differences can still be found, such as:
 
-* `safe_transfer_from` and `safe_batch_transfer_from` are specified such that the optional `data` argument should be of type `Span<felt252>`.
-* `IERC1155Receiver` compliant contracts must return a hardcoded `felt252` representing the SRC5 interface
-ID of `IERC1155Receiver`.
-* `IERC1155Receiver` compliant contracts must support SRC5 by registering the `IERC1155Receiver` interface ID in its constructor and exposing the `supports_interface` method.
+* `safe_transfer_from` and `safe_batch_transfer_from`'s optional `data` argument is implemented as `Span<felt252>`.
+* `IERC1155Receiver` compliant contracts must implement SRC5 and register the `IERC1155Receiver` interface ID.
+* `IERC1155Receiver::on_erc1155_received` must return that interface ID on success.
 In doing so, recipient contracts (both accounts and non-accounts) can be verified that they support ERC1155 transfers.
```

Comment:
i think we can drop this line, not sure it adds much, but no strong opinions if you like it

Reply:
Which line?

Reply:
> In doing so, recipient contracts (both accounts and non-accounts) can be verified that they support ERC1155 transfers.